platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/transients/TransientFieldValueType.java

```diff
@@ -12,7 +12,7 @@
 	public TransientFieldValueType(Class<V> valueClass) {
 
 		this.valueClass = Objects.requireNonNull(valueClass);
-		this.comparator = Optional.empty();
+		this.comparator = Optional.ofNullable(getNaturalOrderComparatorOrNull(valueClass));
```

Comment:
We now use a natural-order `Comparator` for `Comparable` value types, where we used none before.
Shouldn't this change have caused a test to fail?

Reply:
lots of tests are missing, so creating demo module and then tests

Reply:
as discussion, implemented new DemoModule and added tests for sorting and filtering transient field columns